src/NHapi.Base/Parser/EncodingCharacters.cs

```diff
@@ -25,188 +25,187 @@
-		{
-			StringBuilder ret = new StringBuilder();
-
-			for (int i = 0; i < encChars.Length; i++)
-			{
-				ret.Append(encChars[i]);
-			}
-
-			return ret.ToString();
-		}
-
-		public virtual Object Clone()
-		{
-			return new EncodingCharacters(this);
-		}
-
-		public override bool Equals(Object o)
-		{
-			if (o is EncodingCharacters)
-			{
-				EncodingCharacters other = (EncodingCharacters)o;
-				if (FieldSeparator == other.FieldSeparator && ComponentSeparator == other.ComponentSeparator &&
-					 EscapeCharacter == other.EscapeCharacter && RepetitionSeparator == other.RepetitionSeparator &&
-					 SubcomponentSeparator == other.SubcomponentSeparator)
-				{
-					return true;
-				}
-				else
-				{
-					return false;
-				}
-			}
-			else
-			{
-				return false;
-			}
-		}
-
-		public override int GetHashCode()
-		{
-			return 7 * (int)ComponentSeparator * (int)EscapeCharacter * (int)FieldSeparator * (int)RepetitionSeparator *
-					 (int)SubcomponentSeparator;
-		}
-	}
+    using System;
+    using System.Text;
+
+    /// <summary>
+    /// Represents the set of special characters used to encode traditionally
+    /// encoded HL7 messages.
+    /// </summary>
+    /// <author>Bryan Tripp (bryan_tripp@sourceforge.net).</author>
+    public class EncodingCharacters : object, ICloneable
+    {
+        private char fieldSep;
+        private char[] encChars;
+
+        /// <summary>
+        /// Creates new EncodingCharacters object with the given character
+        /// values. If the encodingCharacters argument is null, the default
+        /// values are used.
+        /// </summary>
+        /// <param name="fieldSeparator">The field separator.</param>
+        /// <param name="encodingCharacters">
+        /// Consists of the characters that appear in
+        /// MSH-2 (see section 2.8 of the HL7 spec). The characters are
+        /// Component Separator, Repetition Separator, Escape Character, and
+        /// Subcomponent Separator (in that order).
+        /// </param>
+        public EncodingCharacters(char fieldSeparator, string encodingCharacters)
+        {
+            fieldSep = fieldSeparator;
+
+            encChars = new char[4];
+
+            if (encodingCharacters == null)
+            {
+                encChars[0] = '^';
+
+                encChars[1] = '~';
+
+                encChars[2] = '\\';
+
+                encChars[3] = '&';
+            }
+            else
+            {
+                SupportClass.GetCharsFromString(encodingCharacters, 0, 4, encChars, 0);
+            }
+        }
+
+        public EncodingCharacters(
+            char fieldSeparator,
+            char componentSeparator,
+            char repetitionSeparator,
+            char escapeCharacter,
+            char subcomponentSeparator)
+            : this(
+                fieldSeparator,
+                Convert.ToString(componentSeparator) + repetitionSeparator + escapeCharacter + subcomponentSeparator)

```

Comment:
```csharp
new string(new [] {componentSeparator, repetitionSeparator, escapeCharacter, subcomponentSeparator})
```

I think thats better although i dont know why as it split again in the constructor anyway... seem bzar 😁

Reply:
could just use interpolation right?  Might be better to swap the constructors, add a new one that accepts a `char[]` and mark the one that expects a string as obsolete?

Reply:
Or do what you said but don't mark as obsolete?

Reply:
gone with the first option for now.